File: code_samples/csharp/web_development_middleware.cs
using System;

/// <summary>
/// Web Development: Middleware
/// AI/ML Training Sample
/// </summary>
public class Middleware
{
    private string data;
    
    public Middleware()
    {
        this.data = string.Empty;
    }
    
    public void Process(string input)
    {
        this.data = input;
    }
    
    public string GetData()
    {
        return this.data;
    }
    
    public bool Validate()
    {
        return !string.IsNullOrEmpty(this.data);
    }
    
    public static void Main(string[] args)
    {
        var instance = new Middleware();
        instance.Process("example");
        Console.WriteLine($"Data: {instance.GetData()}");
        Console.WriteLine($"Valid: {instance.Validate()}");
    }
}

File: code_samples/go/web_development_middleware.go
package main

import (
    "fmt"
)

// Web Development: Middleware
// AI/ML Training Sample

type Middleware struct {
    Data string
}

func NewMiddleware() *Middleware {
    return &Middleware{
        Data: "",
    }
}

func (s *Middleware) Process(input string) {
    s.Data = input
}

func (s *Middleware) Validate() bool {
    return len(s.Data) > 0
}

func (s *Middleware) GetData() string {
    return s.Data
}

func main() {
    instance := NewMiddleware()
    instance.Process("example")
    fmt.Printf("Data: %s\n", instance.GetData())
    fmt.Printf("Valid: %v\n", instance.Validate())
}
